### DISCUSSION TITLE
A custom formatter with the python version

### DISCUSSION THREAD

**Discussion by user-name-name** [mitsuhiko/minijinja#237] [Q&A] [answered]
Good day, everyone!

I have a small question regarding using a custom formatter. I had a custom implementation (not the best one) of the python bindings (till Armin has issued his own version minijinja-py) and it had the following formatter for the static Env:

```rust
    ...

    // replace all None values with an empty string instead `none`
    env.set_formatter(|out, state, value| {
        escape_formatter(
            out,
            state,
            if value.is_none() {
                &Value::UNDEFINED
            } else {
                value
            },
        )
    });

    ...
```

So, now I'm thinking how can I add the same behaviour to minijinja-py without forking?

**Comment by mitsuhiko** [comment, marked as answer]
Not yet, but this change will make it work: https://github.com/mitsuhiko/minijinja/pull/238

```python
from minijinja import Environment

def finalizer(value):
    if value is None:
        return ""
    return NotImplemented  # use the default behavior

env = Environment(finalizer=finalizer)
```

**Reply by user-name-name** [reply]
wow, such a speed, thanks a lot :)
